docs/sdk/src/reference_docs/wasm_meta_protocol.rs

```diff
@@ -90,24 +94,66 @@
-//! has and such are all only comprehensible to the runtime.
+//! has and such, are all only comprehensible to the runtime.
 #![doc = simple_mermaid::mermaid!("../../../mermaid/state.mmd")]
 //!
 //! In the above diagram, all of the state keys and values are opaque bytes to the node. The node
-//! does not know what they mean, and it does not now what is the type of the corresponding value
+//! does not know what they mean, and it does not know what is the type of the corresponding value
 //! (e.g. if it is a number of a vector). Contrary, the runtime knows both the meaning of their
 //! keys, and the type of the values.
 //!
-//! This opaque-ness is the fundamental reason why Substrate-based chains can fork-less-ly upgrade:
-//! because the node side code is kept oblivious to all of the details of the state transition
-//! function. Therefore, the state transition function can freely upgrade without the node needing
-//! to know.
+//! This opaque-ness is the fundamental reason why Substrate-based chains can fork-less-ly
+//! upgrade: because the node side code is kept oblivious to all of the details of the state
+//! transition function. Therefore, the state transition function can freely upgrade without the
+//! node needing to know.
 //!
 //! ## Native Runtime
 //!
-//! TODO
+//! Historically, the node software also kept a native copy of the runtime at the time of
+//! compilation within it. This used to be called the "Native Runtime". The main purpose of the
+//! native runtime used to be leveraging the faster execution time and better debugging
+//! infrastructure of native code. However, neither of the two arguments strongly hold and the
+//! native runtime is being fully removed from the node-sdk.
 //!
+//! See: <https://github.com/paritytech/polkadot-sdk/issues/62>
+//!
+//! > Also, note that the flags [`sc_cli::ExecutionStrategy::Native`] is already a noop and all
+//! > chains built with Substrate only use WASM execution.
+//!
+//! ### Runtime Versions
+//!
+//! An important detail of the native execution worth learning about is that the node software,
+//! obviously, only uses the native runtime if it is the same code as with the wasm blob stored
+//! onchain. Else, nodes who run the native runtime will come to a different state transition. How
+//! do nodes determine if two runtimes are the same? Through the very important
+//! [`sp_version::RuntimeVersion`]. All runtimes expose their version via a runtime api
+//! ([`sp_api::Core::version`]) that returns this struct. The node software, or other applications,
+//! inspect this struct to examine the identity of a runtime, and to determine if two runtimes are
+//! the same. Namely, [`sp_version::RuntimeVersion::spec_version`] is the main key that implies two
+//! runtimes are the same.
+//!
+//! Therefore, it is utmost important to make sure before any runtime upgrade, the spec version is
+//! updated.
+// TODO: is this mentioned in the runtime upgrade ref doc?
```

Comment:
It's mentioned very briefly here https://github.com/paritytech/polkadot-sdk/blob/d733c77ee2d2e8e2d5205c552a5efb2e5b5242c8/docs/sdk/src/reference_docs/frame_runtime_upgrades_and_migrations.rs#L32-L33

Maybe the `sp_version::RuntimeVersion` docs should be made a bit richer, explaining more what it's used for?